apiary-metastore-auth/src/test/java/com/expedia/apiary/extensions/readonlyauth/listener/ApiaryReadOnlyAuthPreEventListenerTest.java

```diff
@@ -49,87 +46,116 @@
     when(context.getTable()).thenReturn(table);
 
-    readOnlyAuth.onEvent(context);
+    listener.onEvent(context);
   }
 
-  @Test(expected = InvalidOperationException.class)
-  public void onCreateTableEvent() throws MetaException, NoSuchObjectException, InvalidOperationException {
-    PreCreateTableEvent context = mock(PreCreateTableEvent.class);
-    when(context.getEventType()).thenReturn(PreEventType.CREATE_TABLE);
+  @Test
+  public void onReadDatabaseEventAuthorized() throws Exception {
+    PreReadDatabaseEvent context = mock(PreReadDatabaseEvent.class);
+    when(context.getEventType()).thenReturn(PreEventType.READ_DATABASE);
 
-    Table table = new Table();
-    table.setTableName(tableName);
-    table.setDbName(databaseName);
+    Database db = new Database();
+    db.setName(authorizedDatabaseName2);
+    when(context.getDatabase()).thenReturn(db);
 
-    readOnlyAuth.onEvent(context);
+    listener.onEvent(context);
   }
 
   @Test(expected = InvalidOperationException.class)
-  public void onReadDatabaseEvent1() throws MetaException, NoSuchObjectException, InvalidOperationException {
+  public void onReadDatabaseEventUnauthorized() throws Exception {
     PreReadDatabaseEvent context = mock(PreReadDatabaseEvent.class);
     when(context.getEventType()).thenReturn(PreEventType.READ_DATABASE);
 
     Database db = new Database();
-    db.setName(databaseName);
+    db.setName(unauthorizedDatabaseName);
     when(context.getDatabase()).thenReturn(db);
 
-    readOnlyAuth.onEvent(context);
+    listener.onEvent(context);
   }
 
-  @Test
-  public void onReadDatabaseEvent2() throws MetaException, NoSuchObjectException, InvalidOperationException {
-    PreReadDatabaseEvent context = mock(PreReadDatabaseEvent.class);
-    when(context.getEventType()).thenReturn(PreEventType.READ_DATABASE);
+  @Test(expected = InvalidOperationException.class)
+  public void onCreateTableEvent() throws Exception {
+    PreCreateTableEvent context = mock(PreCreateTableEvent.class);
+    when(context.getEventType()).thenReturn(PreEventType.CREATE_TABLE);
 
-    Database db = new Database();
-    db.setName("db2");
-    when(context.getDatabase()).thenReturn(db);
+    Table table = new Table();
+    table.setTableName(tableName);
+    table.setDbName(authorizedDatabaseName1);
 
-    readOnlyAuth.onEvent(context);
+    listener.onEvent(context);
   }
 
   @Test(expected = InvalidOperationException.class)
-  public void onCreateDatabaseEvent() throws MetaException, NoSuchObjectException, InvalidOperationException {
+  public void onCreateDatabaseEvent() throws Exception {
     PreCreateDatabaseEvent context = mock(PreCreateDatabaseEvent.class);
     when(context.getEventType()).thenReturn(PreEventType.CREATE_DATABASE);
 
     Database db = new Database();
-    db.setName(databaseName);
+    db.setName(authorizedDatabaseName1);
+
+    listener.onEvent(context);
+  }
+
+  @Test
+  public void dbListWithSpaceSeparators() throws Exception {
```

Comment:
rename? -> `dbListNamesTrimmed`
The separator is still a comma

Reply:
yeah, the separator is a space and a comma, your suggested name is good though, will change.